src/hypergraph.jl

```diff
@@ -323,6 +323,22 @@ function clean!(h)
 	h
 end
 
+"""
+    prune_hypergraph(h::Hypergraph)
+Returns a pruned copy of `h`, removing all vertices with degree 0 and all hyperedges of size 0.
+"""
+
+function prune_hypergraph(h)
+    hcp = deepcopy(h)
```

Comment:
Both `prune_hypergraph` and `prune_hypergraph!` have an identical body and hence I think that the code of this entire function can be shortened to:
```julia
function prune_hypergraph(h::Hypergraph)
    prune_hypergraph!(deepcopy(h))
end
```
Note that I also added the type of argument since this function makes only sense for a `Hypergraph` argument.

Reply:
did